### ISSUE TITLE
StructuredValue isn't converted to json format

### ISSUE THREAD

**Issue by JuanCarlosSecmotic** [telefonicaid/iotagent-ul#587]
When I create a device with an Object type attribute and send values ​​to iotagent-ul, these values ​​are converted to an object without problem and stored in orion as an object, but the same does not happen when using the StructuredValue type.
Example:
I create this device:
```
{
    "devices": [
        {
            "device_id": "test",
            "entity_name": "test_name",
            "entity_type": "test_type",
            "transport": "HTTP",
            "attributes": [
                {
                    "name": "t",
                    "type": "StructuredValue"
                },
                                {
                    "name": "z",
                    "type": "Object"
                }
            ]
        }
    ]
}
```
When I send this object to the z attribute, the api responds 200 ok:
```
z|{"test_key":"1"}
```
But when I send the attribute t the same object I get an error:
```
{
    "name": "ENTITY_GENERIC_ERROR",
    "message": "Error accesing entity data for device: test_name of type: test_type"
}
```
This is the message that appears in the IoT logs:

```
time=2023-03-07T06:53:03.594Z | lvl=DEBUG | corr=1b623aa6-37a5-420d-b9e2-a9c879aeed51 | trans=1b623aa6-37a5-420d-b9e2-a9c879aeed51 | op=IoTAgentNGSI.Request | from=n/a | srv=n/a | subsrv=n/a | msg=Options: {
    "url": "http://orion:1026/v2/entities/test_name/attrs?type=test_type",
    "method": "PATCH",
    "headers": {
        "fiware-service": "test",
        "fiware-servicepath": "/test"
    },
    "json": {
        "t": {
            "type": "StructuredValue",
            "value": "{\"test_key\":\"1\"}",
            "metadata": {
                "TimeInstant": {
                    "type": "DateTime",
                    "value": "2023-03-07T06:53:03.594Z"
                }
            }
        },
        "TimeInstant": {
            "type": "DateTime",
            "value": "2023-03-07T06:53:03.594Z"
        }
    }
} | comp=IoTAgent
time=2023-03-07T06:53:03.598Z | lvl=DEBUG | corr=1b623aa6-37a5-420d-b9e2-a9c879aeed51 | trans=1b623aa6-37a5-420d-b9e2-a9c879aeed51 | op=IoTAgentNGSI.Request | from=n/a | srv=n/a | subsrv=n/a | msg=Response {
    "error": "BadRequest",
    "description": "Invalid characters in attribute value"
} | comp=IoTAgent
```

Perhaps the function called "castJsonNativeAttributes" should consider that if the type is StructuredValue it treats it equal to an Array or an Object.

**Comment by mrutid** [comment]
the preferred way to address the casting scenarios is to explicitly transform the measures by using JEXL expression. Use the "autocast" feature is discouraged. Probably It's going to be deprecated soon.

In this way try to do the following provision:

{
    "devices": [
        {
            "device_id": "test",
            "entity_name": "test_name",
            "entity_type": "test_type",
            "transport": "HTTP",
            "attributes": [
                {
                    "name": "t",
                    "type": "StructuredValue",
                    "expression":"t|jsonparse"
                },
                                {
                    "name": "z",
                    "type": "Object"
                }
            ]
        }
    ]
}

you may use a more sofisticated way to do the casting like:

"expression":"(z|typeof)=='string'?z|jsonparse:z"

Remark that you have the following [JEXL transformations] avalible (https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/expressionLanguage.md#available-functions):
...
    jsonparse: (str) => JSON.parse(str),
    jsonstringify: (obj) => JSON.stringify(obj),
    typeof: (val) => typeof val,
    isarray: (arr) => Array.isArray(arr),
    isnan: (val) => isNaN(val),
    parseint: (val) => parseInt(val),
    parsefloat: (val) => parseFloat(val),
    toisodate: (val) => new Date(val).toISOString(),
    tostring: (val) => val.toString(),
 ...

**Comment by mrutid** [comment]
Moreover... in the meanwhile we are going to fix the current autocast https://github.com/telefonicaid/iotagent-node-lib/issues/1323

So we will make it independent from the NGSI type (that doesn't have a concrete or standart meaning) and it will be centered only in the value itself.

**Comment by fgalan** [comment]
@JuanCarlosSecmotic we are in the process of intensive refactoring the IOTA Lib. I'd suggest to wait until that refactor ends (a new IOTA Lib version 3.0.0 will be released and new IOTA-JSON and IOTA-UL versions 2.0.0 will be released using them) to test again and re-validate (or close) the issue.

**Comment by AlvaroVega** [comment]
Currently is in iotagent-ul after PR https://github.com/telefonicaid/iotagent-ul/pull/641 you can do the following:

curl -i -X GET 'http://localhost:7896/iot/d?i=test&d=t|\{"test_key":1\}&k=<apikey>'

curl -i -X GET 'http://localhost:7896/iot/d?i=test&d=z|\{"test_key":1\}&k=<apikey>'


and both t an z are created in entity as json value.
